packages/components/src/date-time/date.js

```diff
@@ -1,24 +1,125 @@
 /**
  * External dependencies
  */
-import ReactDatePicker from 'react-datepicker';
+import classnames from 'classnames';
 import moment from 'moment';
+import { DayPickerSingleDateController } from 'react-dates';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
 
 /**
  * Module Constants
  */
 const TIMEZONELESS_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
 
-function DatePicker( { currentDate, onChange, ...args } ) {
-	const momentDate = currentDate ? moment( currentDate ) : moment();
-	const onChangeMoment = ( newDate ) => onChange( newDate.format( TIMEZONELESS_FORMAT ) );
+class DatePicker extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			descriptionVisible: false,
+			randomId: Math.random().toString().substr( 2 ),
```

Comment:
It's to make sure the date picker help overlay has a unique ID. The picker may be used elsewhere, so instead of `datepicker-container__aria-description`, it has a unique identifier the screen reader may refer to.